File: Tree Height of a Binary Tree.c


/* you only have to complete the function given below.  
node is defined as  

struct node {
    
    int data;
    struct node *left;
    struct node *right;
  
};

*/
int getHeight(struct node* root)
 {
     if(root==NULL)
     {
         return -1;
     }
     int lh=getHeight(root->left);
     int rh=getHeight(root->right);
     if (lh > rh) {
       return lh + 1;
     } 
     else
     {
         return rh+1;
     }
    // Write your code here
}

File: Tree Height of a Binary Tree.java


    /*
    class Node 
        int data;
        Node left;
        Node right;
    */
    public static int height(Node root) 
    {
        if(root==null)
        {
            return -1;
        }

        int ldepth=height(root.left);
        int rdepth=height(root.right);
        if(ldepth>rdepth)
        {
            return (ldepth+1);
        }
        else
        {
            return (rdepth+1);
        }
    }

